### PR TITLE
Add rust formatting

### PR DESCRIPTION
**Pull request by ghost**
# Add rust formatting
Add rust formatting + best practices page

## Changes

- Added rust formatting + rustfmt.toml
- Added best practices page

### CODE REVIEW

File: style/rustfmt.toml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+error_on_unformatted = true
+imports_granularity = "Crate"
+tab_spaces = 4
+hard_tabs = false
+indent_style = "Block"
+max_width = 100
```
Comment (Nomalah) [inline style/rustfmt.toml]: Change max_width to 150 to match other formatting files

File: style/StyleGuide.md [hunk 2]
```diff
@@ -39,6 +39,18 @@ def snake_case(snake_case):
...
 
+### Rust
+```rust
+struct PascalCase {};
+enum PascalCase {
+    PascalCase1,
+    PascalCase2
+};
+trait PascalCase {};
+fn snake_case(snake_case: snake_case) -> snake_case {};
```
Comment (Nomalah) [inline style/StyleGuide.md]: I think you mean:
```suggestion
fn snake_case(snake_case: PascalCase) -> PascalCase {
	return snake_case; // Explicit returns only
};
```

**Review by Nomalah — CHANGES_REQUESTED** [review]
Looks pretty good overall.

File: style/StyleGuide.md (hunk 2 above)
Comment (ghost) [reply style/StyleGuide.md]: Woops yeah

File: style/rustfmt.toml (hunk 1 above)
Comment (ghost) [reply style/rustfmt.toml]: A bit hesitant on this. Many IDEs such as CLion realistically only support 100 characters of width before it goes off screen.

**Pull request merged**